config/initializers/sidekiq_hacks.rb

```diff
@@ -0,0 +1,85 @@
+# frozen_string_literal: true
+
+# This monkey path can be deleted after migrating to Sidekiq >= 7
+module OptionsPatch
+  def client_opts(options)
+    valid_keys = %i[url host port namespace sentinels role path username password ssl_params]
+    super(options.slice(*valid_keys))
+  end
+end
+
+Sidekiq::RedisConnection::RedisAdapter.prepend OptionsPatch
```

Comment:
I'm not a fan of monkey patches, but this is required to fix a bug in Sidekiq, due to the new `redis-client` gem we use not being officially supported until `sidekiq >= 7`

Reply:
A comment will help here too.